src/main/kotlin/org/dif/DIDComm.kt

```diff
@@ -157,6 +159,12 @@ class DIDComm(private val didDocResolver: DIDDocResolver, private val secretReso
         val secretResolver = params.secretResolver ?: this.secretResolver
         val recipientKeySelector = RecipientKeySelector(didDocResolver, secretResolver)
 
-        return unpack(params, recipientKeySelector)
+        try {
+            return unpack(params, recipientKeySelector)
+        } catch (e: IllegalArgumentException) {
+            throw MalformedMessageException(e.localizedMessage, e)
```

Comment:
As I understand this PR only about tests? will some echesment of exception handling be implemented separately?